.github/workflows/publish.yml

```diff
@@ -0,0 +1,55 @@
+# This github action builds Alibi Detect and publishes it to PyPI.
+# It is only run when a new tag is pushed to github.
+# See https://packaging.python.org/en/latest/guides/publishing-package-distribution-releases-using-github-actions-ci-cd-workflows/
+name: Publish 📦 to PyPI 
+
+# Only run when tags starting with "v" are pushed
+on: 
+  push:
+    tags:
+      - 'v*'
+
+jobs:
+  build-and-publish:
+    name: Build and publish
+    runs-on: ubuntu-18.04
+
+    steps:
+    # Setup version extractor (from tag)
+    - uses: nowsprinting/check-version-format-action@v3
+      id: version
+      with:
+        prefix: 'v'
+
+    # Setup Python
+    - uses: actions/checkout@master
+    - name: Set up Python 3.9
+      uses: actions/setup-python@v1
+      with:
+        python-version: 3.9
+
+    # Build
+    - name: Install pypa/build
+      run: python -m pip install --user build==0.9.0
+    - name: Build a binary wheel and a source tarball
+      run: python -m build --sdist --wheel --outdir dist/ .
+
+    # Publish to Test PyPI unconditionally
+    - name: Publish 📦 to Test PyPI
+      uses: pypa/gh-action-pypi-publish@release/v1
+      with:
+        password: ${{ secrets.TEST_PYPI_API_TOKEN }}
+        repository_url: https://test.pypi.org/legacy/
+
+# TODO - Test below by:
```

Comment:
@jklaise re testing the real PyPI release process. I think these steps should work. The thing we want to avoid is a a non-stable tag triggering a release, so we want to test the `if: steps.version.outputs.prerelease == ''` line. As long as we try out the below before adding a `PYPI_API_TOKEN` to the `SeldonIO/alibi-detect` repo we should be able to test with no risk of accidental release.

Reply:
p.s. just realised I am being silly here. We can test the condition bit by just adding a temporary `echo`...